src/python/gem5/components/boards/arm_board.py

```diff
@@ -211,6 +222,11 @@ def _setup_io_devices(self) -> None:
         if hasattr(self.realview.gic, "cpu_addr"):
             self.gic_cpu_addr = self.realview.gic.cpu_addr
 
+        # For KVM cpus, we need to simulate the GIC.
+        if any(core.is_kvm_core() for core in self.processor.get_cores()):
+            GenericTimer.generateDeviceTree = SimObject.generateDeviceTree
```

Comment:
This is *very* odd. What are you doing in this line? It looks like you're modifying the `GenericTimer` python class. This is a bad code smell.

Reply:
So, as far as I understand, we cannot rely on the host's generic timer when we want to use KVM in ARM and simulate the GIC in gem5. This is taken from the gem5's ARM FS tests. 
The comment on the test file (`tests/fs/linux/arm/configs/arm_generic.py`) states the following:
```py
        # Arm KVM regressions will use a simulated GIC. This means that in
        # order to work we need to remove the system interface of the
        # generic timer from the DTB and we need to inform the MuxingKvmGic
        # class to use the gem5 GIC instead of relying on the host one
```

Reply:
I see.. it's still bad code. You're taking the member function `GenericTimer.generateDeviceTree` and *replacing the function` with `SimObject.generateDeviceTree`.

This shows a big problem with creating instances of SimObject in the SimObject description files. I guess I'll let this slide since it's already there, but it's not good code practice.

**Please copy over this comment, though!**